lib/ring_buffer/ring_buffer.c

```diff
@@ -79,16 +80,136 @@ int ring_buf_item_get(struct ring_buf *buf, u16_t *type, u8_t *value,
 	if (likely(buf->mask)) {
 		for (i = 0; i < header->length; ++i) {
 			index = (i + buf->head + 1) & buf->mask;
-			data[i] = buf->buf[index];
+			data[i] = buf->buf.buf32[index];
 		}
 		buf->head = (buf->head + header->length + 1) & buf->mask;
 	} else {
 		for (i = 0; i < header->length; ++i) {
 			index = (i + buf->head + 1) % buf->size;
-			data[i] = buf->buf[index];
+			data[i] = buf->buf.buf32[index];
 		}
 		buf->head = (buf->head + header->length + 1) % buf->size;
 	}
 
 	return 0;
 }
+
+/** @brief Wraps if exceeds the limit.
```

Comment:
Another pretty quick a description, even for brief. Maybe at least "Wrap index if it exceeded the limit"?